plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -633,6 +651,36 @@ public WorkspaceEdit organizeImports(String path) {
     return doGetOne("java.edit.organizeImports", singletonList(prefixURI(path)), type);
   }
 
+  // configuration
+
+  /**
+   * Returns JDT LS java core options.
+   *
+   * @param filter list of the specific options to return.
+   */
+  public JavaCoreOptions getJavaCoreOptions(List<String> filter) {
+    Type type = new TypeToken<JavaCoreOptions>() {}.getType();
+    return doGetOne(Commands.GET_JAVA_CORE_OPTIONS_СOMMAND, new ArrayList<>(filter), type);
+  }
+
+  /** Returns JDT LS preferences */
+  public JdtLsPreferences getJdtLsPreferences() {
+    Type type = new TypeToken<JdtLsPreferences>() {}.getType();
+    return doGetOne(Commands.GET_PREFERENCES_СOMMAND, emptyList(), type);
+  }
+
+  /** Updates JDT LS java core options. */
+  public Boolean updateJavaCoreOptions(JavaCoreOptions javaCoreOptions) {
+    Type type = new TypeToken<Boolean>() {}.getType();
+    return doGetOne(
+        Commands.UPDATE_JAVA_CORE_OPTIONS_СOMMAND, singletonList(javaCoreOptions), type);
+  }
+  /** Updates JDT LS preferences. */
+  public Boolean updatePreferences(JdtLsPreferences preferences) {
+    Type type = new TypeToken<Boolean>() {}.getType();
+    return doGetOne(Commands.UPDATE_PREFERENCES_СOMMAND, singletonList(preferences), type);
```

Comment:
Is "update preferences" called anywhere? If not, let's not add commands that we don't use.

Reply:
ok